pkg/controller/certificates/signer/signer.go

```diff
@@ -111,13 +94,65 @@ func (s *signer) handle(csr *capi.CertificateSigningRequest) error {
 	return nil
 }
 
+// maybeReloadCA stores the current cert/key content if the values have changed
+func (s *signer) maybeReloadCA() error {
+	certPEM, keyPEM := s.caLoader.CurrentCertKeyContent()
+	currCA := s.currentCA()
+	if bytes.Equal(currCA.RawCert, certPEM) && bytes.Equal(currCA.RawKey, keyPEM) {
+		return nil
+	}
+
+	return s.setCA(s.caLoader)
+}
+
+// setCA unconditionally stores the current cert/key content
+func (s *signer) setCA(certKeyContent dynamiccertificates.CertKeyContentProvider) error {
+	certPEM, keyPEM := certKeyContent.CurrentCertKeyContent()
+
+	certs, err := cert.ParseCertsPEM(certPEM)
+	if err != nil {
+		return fmt.Errorf("error reading CA cert file %q: %v", certKeyContent.Name(), err)
+	}
+	if len(certs) != 1 {
+		return fmt.Errorf("error reading CA cert file %q: expected 1 certificate, found %d", certKeyContent.Name(), len(certs))
```

Comment:
Could you please add support for CA bundle here?
Namely, gracefully go through the list of `certs` and find the certificate that corresponds to `key`, instead of throwing the error?

This can be done similar to [FindCertForKey function](https://github.com/kubernetes/kubernetes/pull/86833/files#diff-99a17a50a4fb71b44af8fc6ae82409f9R516).

I could also submit a separate patch for CA bundle support in controller-manager if this change is out of scope of this PR.

HDYT, @mikedanese?

This change will help to merge CA bundle support in `kubeadm`...

Reply:
> Could you please add support for CA bundle here?
> Namely, gracefully go through the list of `certs` and find the certificate that corresponds to `key`, instead of throwing the error?

I do not plan to change the behavior of parsing this file in this PR.  Based on my read of the doc for this argument, I didn't expect it to be a bundle.

Reply:
Supporting intermediate CAs is a significant feature addition, and we would need to consider it carefully. I'm open to the idea but it's not related to this change. If there's not a feature request tracking this, feel free to open one.